test/helpers/timed-out.ts

```diff
@@ -0,0 +1,71 @@
+import {RequestOptions} from 'http';
+import http = require('http');
+import https = require('https');
+import lolex = require('lolex');
+import {kTimers} from '../../source/utils/timed-out';
+
+export function init() {
+	const clock = lolex.createClock();
+	let requestWasCreated = false;
+
+	return {
```

Comment:
Could you please help me understand this code? I just don't know what's this responsible for.

Reply:
I think I get it. You don't override the global `setTimeout`, so you have to override `request.setTimeout`. But still, this looks way too complicated.